### ISSUE TITLE
Labeling

### ISSUE THREAD

**Issue by kimberly-moors** [kimberly-moors/Facebook-emoticons#3]
Ik zit volledig vast mij het labelen van mijn data. 
Ik zou graag cEXT is 'y' of 'n' vervangen door labels: O (introvert) en 1 (extravert).
Moet ik hier een nieuwe kolom voor aanmaken of kan ik dit rechtstreeks in mijn kolom doen?
De push "labelling" bevat alles wat ik tot nu toe heb ivm dit topic

**Comment by walterdaelemans** [comment]
sklearn gebruikt inderdaad 0 en 1 maar je kan de LabelEncoder gebruiken daarvoor
le = LabelEncoder()
y = le.fit_transform(data.kolom)  
\# y is nu een kolom met 1 en 0
\# data.kolom is de kolom in je pandas dataframe waar de klassen in staan
le.classes_
le.transform(['y','n'])   -> array([0, 1])

**Comment by stephantul** [comment]
Een ander probleem is dat je nu je index aan `y`  toewijst. Als je dit doet, dan krijgt elke instance eigenlijk een uniek label. In plaats daarvan kan je bijvoorbeeld het volgende doen:
```
X = df.STATUS
y = df.cEXT
```
Als je nu `y` aan een LabelEncoder geeft, zoals Walter al zei, dan zullen deze labels automatisch categoriale numerieke variabelen worden (bijv. 0 en 1 als je maar 2 waarden hebt.)

**Comment by kimberly-moors** [comment]
Bedankt dit is gelukt. 
Ik kom nu een set van nullen en eentjes uit. 
Bij het ingeven van:
     <<le.fit_transform(dataset.column)>>
Kreeg ik echter steeds de melding dat ik een positional argument miste. Dit heb ik uiteindelijk opgelost door er (dataset.colum,y) van te maken.

**Comment by stephantul** [comment]
Dit wordt veroorzaakt doordat je je labelencoder niet initialiseert.
Dit lost dat op: ```encoded_labels = le().fit_transform(labels)```

**Comment by stephantul** [comment]
Je hebt het LabelEncoder object niet geinitialiseerd (dat doe je met `le = LabelEncoder()`, zoals walter hierboven al postte.) Wat je nu doet werkt misschien wel, maar is niet juist, en ik weet niet welke resultaten dat oplevert.

**Comment by kimberly-moors** [comment]
De initialisatie van de LabelEncoder heb ik erboven op deze manier gedaan, is dit dan fout?

`from sklearn.preprocessing import LabelEncoder as le`

**Comment by stephantul** [comment]
Dat is geen initialisatie, dat is een import. In principe zeg je met die lijn gewoon dat je `LabelEncoder` vanaf nu `le` noemt. Daarmee verander je echter de functie van de naam niet.

**Comment by kimberly-moors** [comment]
Bedankt.
